students/gregdevore/lesson02/assignment/code/charges_calc.py

```diff
@@ -0,0 +1,140 @@
+"""
+Returns total price paid for individual rentals
+
+This module supports logging, which can be enabled with an optional
+input argument (-d or --debug). The supported levels are:
+0 - Do not log anything
+1 - Log errors only (to a file with today's date)
+2 - Log errors and warnings only (to a file with today's date)
+3 - Log debug messages (to the console) and errors and warnings
+    (to a file with today's date)
+
+Debug messages - Track the flow of the script
+Warning messages - Identify missing elements in source JSON data
+Error messages - Identify inconsistencies in source JSON data that can cause
+                 the script to crash
+"""
+import argparse
+import json
+import datetime
+import math
+import logging
+
+def parse_cmd_arguments():
+    """
+    Define command line argument structure for input/output JSON files
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
+    # Add optional argument for debugging
+    parser.add_argument('-d', '--debug', type=int,
+                        help='debug level (optional)', required=False)
+
+    return parser.parse_args()
+
+def load_rentals_file(filename):
+    """
+    Load JSON file using python json module
+    """
+    logging.debug('Loading JSON file.')
+    with open(filename) as file:
+        try:
+            data = json.load(file)
+        except Exception:
```

Comment:
Avoid using "catch all" exception handling. The risk with this is that your code could eventually run into a corner condition that could end up being incorrectly handled (and being hidden) by the catch-all, triggering other issues. You should always add handling for specific issues you know/think your code could run into and let it crash if something else happens, so you that proper debug can be done (for some applications, this might not be acceptable, so you could have general exception handling in those cases, making sure the log shows that "something else" happen, so that the issue does not remain unnoticed).